Packs/CapeSandbox/Integrations/CapeSandbox/CapeSandbox.py

```diff
@@ -0,0 +1,1665 @@
+
+    # Basic Auth credentials
+    credentials = params.get("credentials", {})
+    username = credentials.get("identifier", params.get("username"))
+    password = credentials.get("password", params.get("password"))
+
+    demisto.debug(
+        f"Parsed config: {base_url=}, {verify_certificate=}, "
+        f"{proxy=}, has_api_token={bool(api_token)}, has_username={bool(username)}"
+    )
+
+    if not base_url:
+        raise DemistoException("Server URL (url) is required.")
+
+    # precedence: api_token over username/password
+    if api_token and (username or password):
+        demisto.debug("CapeSandbox: api_token provided; username/password will be ignored.")
+
+    return {
+        "base_url": base_url,
+        "verify_certificate": verify_certificate,
+        "proxy": proxy,
+        "api_token": api_token,
+        "username": username,
+        "password": password,
+    }
+
+
+def build_submit_form(args: dict[str, Any], url_mode: bool = False) -> dict[str, Any]:
+    demisto.debug(f"Building submit form with {url_mode=}")
+    form = assign_params(
+        package=args.get("package"),
+        timeout=arg_to_number(args.get("timeout")),
+        priority=arg_to_number(args.get("priority")),
+        options=args.get("options"),
+        machine=args.get("machine"),
+        platform=args.get("platform"),
+        tags=args.get("tags"),
+        custom=args.get("custom"),
+        memory="1" if argToBoolean(args.get("memory", False)) else None,
+        enforce_timeout=("1" if argToBoolean(args.get("enforce_timeout", False)) else None),
+        clock=args.get("clock"),
+    )
+    if url_mode:
+        form["url"] = args.get("url")
+    demisto.debug(f"Built form with {len(form)} parameters: {list(form.keys())}")
+    return form
+
+
+def extract_entry_file_data(entry_id: str) -> tuple[str, str]:
+    """
+    Retrieves the local file path and original name for a given entry ID.
+    Handles all potential errors related to file fetching.
+
+    Args:
+        entry_id: The ID of the entry (e.g., file) in the War Room.
+
+    Returns:
+        A tuple containing (file_path: str, file_name: str).
+
+    Raises:
+        DemistoException: If the entry is not found or is not a file.
+    """
+    try:
+        filepath_result = demisto.getFilePath(entry_id)
+
+        if not filepath_result or "path" not in filepath_result:
+            raise ValueError("Entry is not a valid file entry.")
+
+    except ValueError as error:
+        demisto.debug(f"Error fetching file path for {entry_id!r}", error)
+        raise DemistoException(f"Could not find file or entry: {entry_id!r}")
+
+    except Exception as error:
+        raise DemistoException(f"An unexpected error occurred while processing entry {entry_id!r}", error)
+
+    path = filepath_result["path"]
+    name = filepath_result.get("name")
+
+    final_name = name or os.path.basename(path)
+
+    return path, final_name
+
+
+TYPE_MAP = {
```

Comment:
Is this intended to be here or in the constant segment?
If this is intended to be in a pre-determined format, then you should consider some enforcement such as `namedtuple` or a `class`, as a map is vulnerable for typos.